### DISCUSSION TITLE
Windows: Space(s) in paths don't work?

### DISCUSSION THREAD

**Discussion by highend** [creativeprojects/resticprofile#433] [Q&A]
Hi,

I'm trying to use resticprofile on Windows.
resticprofile 0.29.0
restic 0.17.3

Test ```profiles.toml```:
```
version = "1"

[global]
  restic-binary = "D:\\Tools\\@Command Line Tools\\restic_x64\\restic.exe"

[default]
  repository = "T:\\backup\\restic"
  password-file = "D:\\Tools\\@Command Line Tools\\restic_x64\\key.pub"

  [default.backup]
    verbose = true
    skip-if-unchanged = true
    cache-dir = "T:\\restic_cache"
    files-from = "D:\\Tools\\@Command Line Tools\\restic_x64\\include_paths.txt"
    exclude-file = "D:\\Tools\\@Command Line Tools\\restic_x64\\exclude_patterns.txt"
```

```resticprofile.exe --verbose backup --dry-run```

Result:
```
2024/11/22 00:57:21 resticprofile 0.29.0 compiled with go1.23.2
2024/11/22 00:57:21 using configuration file: profiles.toml
2024/11/22 00:57:21 loading: profiles.toml
2024/11/22 00:57:21 memory available: 4656MB
2024/11/22 00:57:21 using restic 0.17.3
2024/11/22 00:57:21 profile 'default': starting 'backup'
2024/11/22 00:57:21 command environment: reusing previous
2024/11/22 00:57:21 starting command: D:\Tools\@Command Line Tools\restic_x64\restic.exe backup --cache-dir=T:\restic_cache --exclude-file=D:\Tools\@Command Line Tools\restic_x64\exclude_patterns.txt --files-from=D:\Tools\@Command Line Tools\restic_x64\include_paths.txt --password-file=T:\backup\restic\key.pub --repo=T:\backup\restic --skip-if-unchanged --verbose --dry-run
'D:\Tools\@Command' is not recognized as an internal or external command,
operable program or batch file.
2024/11/22 00:57:21 backup on profile 'default': exit status 1
```

If I double quote all paths and run restic with the output of the command from above, the backup runs fine (ofc, with --dry-run here)...

What am I doing wrong?

**Comment by erichlf** [comment]
When you have spaces in a directory name it needs to be quoted. That is the nature of using *nix type scripts.

**Comment by highend** [comment]
Where do you see unquoted paths in my config (for Windows^^)?

**Reply by erichlf** [reply]
I suppose this could be considered a bug, because you shouldn't have to add a second set of quotes to have quoted paths. I would report it under issues.

**Comment by danielfdickinson** [comment]
You need a single backslash (\) in front of spaces. E.g.:

 restic-binary = ***@***.***\ Line\ Tools\\restic_x64\\restic.exe"



________________________________
From: highend
Sent: Thursday, November 21, 2024 7:18 PM
To: creativeprojects/resticprofile
Cc: Subscribed
Subject: [creativeprojects/resticprofile] Windows: Space(s) in paths don't work? (Discussion #433)


Hi,

I'm trying to use resticprofile on Windows.
resticprofile 0.29.0
restic 0.17.3

Test profiles.toml:

version = "1"

[global]
  restic-binary = ***@***.*** Line Tools\\restic_x64\\restic.exe"

[default]
  repository = "T:\\backup\\restic"
  password-file = ***@***.*** Line Tools\\restic_x64\\key.pub"

  [default.backup]
    verbose = true
    skip-if-unchanged = true
    cache-dir = "T:\\restic_cache"
    files-from = ***@***.*** Line Tools\\restic_x64\\include_paths.txt"
    exclude-file = ***@***.*** Line Tools\\restic_x64\\exclude_patterns.txt"


resticprofile.exe --verbose backup --dry-run

Result:

2024/11/22 00:57:21 resticprofile 0.29.0 compiled with go1.23.2
2024/11/22 00:57:21 using configuration file: profiles.toml
2024/11/22 00:57:21 loading: profiles.toml
2024/11/22 00:57:21 memory available: 4656MB
2024/11/22 00:57:21 using restic 0.17.3
2024/11/22 00:57:21 profile 'default': starting 'backup'
2024/11/22 00:57:21 command environment: reusing previous
2024/11/22 00:57:21 starting command: ***@***.*** Line Tools\restic_x64\restic.exe backup --cache-dir=T:\restic_cache ***@***.*** Line Tools\restic_x64\exclude_patterns.txt ***@***.*** Line Tools\restic_x64\include_paths.txt --password-file=T:\backup\restic\key.pub --repo=T:\backup\restic --skip-if-unchanged --verbose --dry-run
***@***.***' is not recognized as an internal or external command,
operable program or batch file.
2024/11/22 00:57:21 backup on profile 'default': exit status 1


If I double quote all paths and run restic with the output of the command from above, the backup runs fine (ofc, with --dry-run here)...

What am I doing wrong?

**Reply by creativeprojects** [reply]
Yes that would work, but I do agree you shouldn't need to.
There are so many places where escaping of path should happen and I missed the restic binary case 😢
